cmd/influx/main.go

```diff
@@ -91,9 +93,42 @@ func newApp() cli.App {
 	}
 }
 
+// This creates a new slice and replaces `-t "-FOO-TOKEN"` with `-t=-FOO-TOKEN`
+// This is necessary to do because the command line arg:
+//  `-t "-FOO-TOKEN"`` will be parsed as two separate flags instead of a flag and token value.
+func replaceTokenArg(args []string) []string {
+	newArgs := make([]string, len(args))
+	copy(newArgs, args)
+	for i, arg := range newArgs {
+		switch arg {
+		case "--token", "-t":
+			// if last element, this will be invalid later
+			if i == len(args)-1 {
+				break
+			}
+			if strings.HasPrefix(newArgs[i+1], "-") {
+				color.HiYellow("warning: %s %s interpreted as %s=%s, consider using %s=%s syntax when tokens start with a hyphen",
+					newArgs[i], newArgs[i+1],
+					newArgs[i], newArgs[i+1],
+					newArgs[i], newArgs[i+1],
+				)
+			}
+			newArgs[i] = strings.Join(newArgs[i:i+2], "=")
+			// if there are 2+ elements after this
+			if len(newArgs) > i+2 {
+				newArgs = append(newArgs[:i+1], newArgs[i+2:]...)
```

Comment:
I don't think you need the 'if' here - `newArgs` would be an empty slice if `len(newArgs) == i+2`